tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalDivisionSuite.scala

```diff
@@ -0,0 +1,287 @@
+          (getSign * r.nextInt(1024 * 1024) + 1).toLong)
+      })
+
+      val schema = StructType(Seq(
+        StructField("c_ym", YearMonthIntervalType()),
+        StructField("c_i", IntegerType),
+        StructField("c_l", LongType)))
+      spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+    }
+  ) {
+    df => df.selectExpr("c_ym / c_i", "c_ym / c_l")
+  }
+
+  // both gpu and cpu will throw ArithmeticException
+  def testDivideYMOverflow(testCaseName: String, months: Int, num: Number): Unit = {
+    testBothCpuGpuExpectedException[SparkException](testCaseName + ", cv / scalar",
+      e => e.getMessage.contains("ArithmeticException"),
+      spark => {
+        val data = Seq(Row(Period.ofMonths(months)))
+        val schema = StructType(Seq(
+          StructField("c1", YearMonthIntervalType())))
+        spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+      }
+    ) {
+      df =>
+        if (num.isInstanceOf[Byte]) {
+          df.selectExpr(s"c1 / cast('$num' as Byte)")
+        } else if (num.isInstanceOf[Short]) {
+          df.selectExpr(s"c1 / cast('$num' as Short)")
+        } else if (num.isInstanceOf[Int]) {
+          df.selectExpr(s"c1 / $num")
+        } else if (num.isInstanceOf[Long]) {
+          df.selectExpr(s"c1 / ${num}L")
+        } else if (num.isInstanceOf[Float]) {
+          if (num.equals(Float.NaN)) {
+            df.selectExpr("c1 / cast('NaN' as float)")
+          } else if (num.equals(Float.PositiveInfinity)) {
+            df.selectExpr("c1 / cast('Infinity' as float)")
+          }
+          else if (num.equals(Float.NegativeInfinity)) {
+            df.selectExpr("c1 / cast('-Infinity' as float)")
+          } else {
+            df.selectExpr(s"c1 / ${num}f")
+          }
+        } else if (num.isInstanceOf[Double]) {
+          if (num.equals(Double.NaN)) {
+            df.selectExpr("c1 / cast('NaN' as double)")
+          } else if (num.equals(Double.PositiveInfinity)) {
+            df.selectExpr("c1 / cast('Infinity' as double)")
+          }
+          else if (num.equals(Double.NegativeInfinity)) {
+            df.selectExpr("c1 / cast('-Infinity' as double)")
+          } else {
+            df.selectExpr(s"c1 / ${num}d")
+          }
+        } else {
+          throw new IllegalStateException("")
+        }
+    }
+
+    testBothCpuGpuExpectedException[SparkException](testCaseName + ", cv / cv",
+      e => e.getMessage.contains("ArithmeticException"),
+      spark => {
+        // Period is the external type of year-month type
+        val (data, schema) =
+          if (num.isInstanceOf[Byte]) {
+            (Seq(Row(Period.ofMonths(months), num.asInstanceOf[Byte])),
+                StructType(Seq(
+                  StructField("c1", YearMonthIntervalType()),
+                  StructField("c2", ByteType)
+                )))
+          } else if (num.isInstanceOf[Short]) {
```

Comment:
Pattern matching makes more sense here, as it's more concise to write and read and removes the need to cast, e.g.:
```suggestion
        val (data, schema) = num match {
          case n: Byte =>
            (Seq(Row(Period.ofMonths(months), n)),
                StructType(Seq(
                  StructField("c1", YearMonthIntervalType()),
                  StructField("c2", ByteType)
                )))
          case n: Short =>
            ...
```

Reply:
done